src/farcasterd/trade_state_machine.rs

```diff
@@ -624,6 +635,14 @@ fn attempt_transition_to_restoring_swapd(
                 public_offer.offer.accordant_blockchain.try_into()?,
                 public_offer.offer.network,
             )?;
+            let listening =
+                if let Err(err) = runtime.listen(runtime.config.get_bind_addr().unwrap()) {
```

Comment:
Handle the `unwrap` here is required IMO, as we are warning if the `listen` fails we should warn the same way if `.get_bind_addr()` fails too, otherwise it is not possible to restart and handle the swap without a connection.

Reply:
Ok, I'll see if I can handle this configuration error and let the user know of it.

Reply:
Fixed in f71102d758af53c12fd843ca2f20eef87bc6701e